wrap console.log; dont log when used as a module

fippo opened the pull request:
removes console.log when used in a module, fixes #68.
Also removes some inconsistent logging and debug in the resolution stuff.

Ideally we would provide modules with a way to set a logger but... maybe later.

KaptenJansson commented on adapter.js:
```diff
@@ -23,6 +23,15 @@ var webrtcDetectedBrowser = null;
+function webrtcLogger() {
+  // suppress console.log output when being included as a module
+  if (typeof module !== 'undefined' ||
+      ((typeof require === 'function') && (typeof define === 'function'))) {
```
Remove parentheses around the 2nd and 3rd typeof operators, i.e:

    (typeof module !== 'undefined' ||
      typeof require === 'function' && typeof define === 'function')

http://google-styleguide.googlecode.com/svn/trunk/javascriptguide.xml?showone=Parentheses#Parentheses

KaptenJansson commented on adapter.js:
```diff
@@ -23,6 +23,15 @@ var webrtcDetectedBrowser = null;
 var webrtcMinimumVersion = null;
 
+function webrtcLogger() {
+  // suppress console.log output when being included as a module
```
Add full stop "."

alvestrand commented on adapter.js:
```diff
@@ -23,6 +23,15 @@ var webrtcDetectedBrowser = null;
+  // suppress console.log output when being included as a module
+  if (typeof module !== 'undefined' ||
+      ((typeof require === 'function') && (typeof define === 'function'))) {
+    return undefined;
```
For my curiosity's sake: Why should this function return anything at all?

fippo replied:
good question, especially since the default return value is undefined anyway.... removed.

jan-ivar replied:
why not invert the if?

alvestrand commented on adapter.js:
```diff
@@ -23,6 +23,15 @@ var webrtcDetectedBrowser = null;
 var webrtcDetectedVersion = null;
 var webrtcMinimumVersion = null;
 
+function webrtcLogger() {
```
Is this a new function in the global namespace? Can we wrap it inside some module object rather than continuing our tradition of polluting the global namespace with our utility functions?

fippo replied:
Along the lines of 
```
webrtcUtils = {
   log: function () { ... }
};
```
? This still pollutes, but at least we can limit this.

alvestrand commented:
I'm not sure removing the logging of constraint rewrites is a Good Idea at this time... can you make that into a separate PR so that we can get Jan-Ivar's comment on that removal separately?

KaptenJansson commented:
Also could you add test that exercises this?

jan-ivar commented:
@alvestrand yeah it might be premature to remove the constraint conversion logging, if only because the conversion is imperfect (it could go further by implementing fallbacks in the advanced array to simulate the ideal algorithm).

fippo commented:
- added the constraint logs back
- introduced webrtcUtils object which currently only contains a log item.

@KaptenJansson testing this is somewhat tricky since the test requires this as a module. Might be possibly by overriding console and looking for an adapter-specific log but...

alvestrand commented:
Can we test this by letting the test function temporarily override console.log and verify that it's not called?

ie let the test be

savelog = console.log
console.log = function(args) { t.fail() }
actionThatCausesLog
console.log = savelog

Doesn't work for async/promise-driven actions, of course.

alvestrand commented:
(note - it's possible that my suggestion doesn't work because tape uses console.log for printing. We can figure that out by trying....)

fippo commented:
I did something evil... WDYT?
```
var logcount = 0;
var c = console.log.bind(console);
console.log = function() {
  logcount++;
  c.apply(c, arguments);
};
var m = require('../adapter.js');
console.log = c;

test('log suppression', function (t) {
  console.log('logcount', logcount);
});
```
When including adapter, console.log should not be called and that can be asserted as logcount === 0
Note that you need the additional .bind(console) when saving console.log.

alvestrand commented:
That sounds twisted enough for a Javascript test :-)
Let's do it! (if it works...)

KaptenJansson commented on adapter.js:
```diff
@@ -22,6 +22,15 @@ var reattachMediaStream = null;
+  log: function() {
+    // suppress console.log output when being included as a module.
+    if (!(typeof module !== 'undefined' ||
+        (typeof require === 'function') && (typeof define === 'function'))) {
```
Remove parentheses around the 2nd and 3rd typeof operators, i.e:

(typeof module !== 'undefined' ||
  typeof require === 'function' && typeof define === 'function')
http://google-styleguide.googlecode.com/svn/trunk/javascriptguide.xml?showone=Parentheses#Parentheses

KaptenJansson commented on test/test.js:
```diff
@@ -5,7 +5,22 @@
+// when used as a module. This temporarily overloads
+// console.log so we can assert this.
+var logcount = 0;
+var saveconsole = console.log.bind(console);
```
Use camelCase, applies below as well.

KaptenJansson commented on test/test.js:
```diff
@@ -5,7 +5,22 @@
+// adapter.js is not supposed to spill console.log
+// when used as a module. This temporarily overloads
+// console.log so we can assert this.
+var logcount = 0;
```
Use camelCase, applies below as well.

fippo commented:
@KaptenJansson done. too bad jscs can't check this (even though there is http://jscs.info/rule/requireCamelCaseOrUpperCaseIdentifiers.html ?)

KaptenJansson commented:
It should be able to and I believe I've actually seen it do so before, filed #84 for this.

KaptenJansson commented:
LGTM
